flycheck.el

```diff
@@ -1945,9 +1970,26 @@ nil otherwise."
                                 (funcall enabled)))))
     (if shall-enable
         (cl-pushnew checker flycheck-enabled-checkers)
-      (cl-pushnew checker flycheck-disabled-checkers))
+      (cl-pushnew checker flycheck-automatically-disabled-checkers))
     shall-enable))
 
+(defun flycheck-reset-enabled-checker (checker)
+  "Reset the `:enabled' test of CHECKER.
+
+Forget that CHECKER has been enabled or automatically disabled
+from a previous `:enabled' test.  Once a checker has been enabled
+or automatically disabled, `flycheck-may-enable-checker' will
+always be constant (t or nil respectively).
+
+If you wish to test the `:enabled' predicate again, you must
+first reset its state using this function."
+  (when (memq checker flycheck-automatically-disabled-checkers)
+    (setq flycheck-automatically-disabled-checkers
+          (remq checker flycheck-automatically-disabled-checkers)))
+  (when (memq checker flycheck-enabled-checkers)
+    (setq flycheck-enabled-checkers
+          (remq checker flycheck-enabled-checkers))))
```

Comment:
I think we don't want that, since we call this function from `flycheck-verify-checker`

Reply:
That's the point though: if a checker has previously passed the `:enabled` test, it has been added to `flycheck-enabled-checkers` and won't be tested again unless we remove it from that list.

Resetting the `:enabled` state thus means removing the checker from these two lists.

Reply:
Oh, you're completely right. I read `disabled` instead of `enabled` here.